hub/scripts.js

```diff
@@ -301,6 +301,45 @@
     }
   }
 
+  /**
+   * Parses the page URL and retrieves the locale, cloud and category.
+   * Page details are updated based on these values.
+   */
+  function handlePageDetails() {
+    const MAP = {
+      ca: { country: 'CA', language: 'en' },
+      ca_fr: { country: 'CA', language: 'fr' },
+      en: { country: 'US', language: 'en' },
+    };
+    const [locale, cloud, category] = window.location.pathname
+      .split('/').filter(isNonEmptyString);
```

Comment:
Is the `isNonEmptyString` really necessary? An URL with a double slash at some point will not be valid. But even if it is, say that the form is `a//c`. `[locale, cloud, category] = window.location.pathname` will mean that `locale === a` and `cloud === 'c'`, while `category === undefined`. When in reality, `locale` and `category` are defined, but the `cloud` is not. Am I missing something?

Reply:
yes, the filter is necessary because I'm using the `location.pathname`, which starts with a `/`; an empty entry will be added in the output array after split. Here's an example:
```
URL - https://www.adobe.com/de/creativecloud/photography/discover.html
pathname is: /de/creativecloud/photography/discover.html
pathname.split - ["", "de", "creativecloud", "photography", "discover.htlm"]
```

Reply:
Ah, yes, forgot about the first slash. Thanks for the reminder 😄